pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -680,6 +680,7 @@ public static class Offline {
     public static class AssignmentStrategy {
       public static final String BALANCE_NUM_SEGMENT_ASSIGNMENT_STRATEGY = "BalanceNumSegmentAssignmentStrategy";
       public static final String REPLICA_GROUP_SEGMENT_ASSIGNMENT_STRATEGY = "ReplicaGroupSegmentAssignmentStrategy";
+      public static final String DIM_TABLE_SEGMENT_ASSIGNMENT_STRATEGY = "DimTableSegmentAssignmentStrategy";
```

Comment:
Our new config doesn't necessarily need to use the same pattern. Alternative way is to have sth like the following:

```
{
  "segmentAssignmentConfigMap": {
    "OFFLINE": {
      "segmentAssignmentStrategy": "balanced/replicaGroup/dimTable"
    }
  }
  ...
}

```

Reply:
also, are we covering the different cases? we can make all of the following to work. `e.g. balanced, Balanced, BaLanCed...etc`

Reply:
+1 on keeping them short and concise. We might not want a strategy name for the dimension table. If we want to make one, we want to make it more general, such as `allServers`

Reply:
updated